src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -4690,25 +4689,25 @@ class StubGenerator: public StubCodeGenerator {
     const Register x     = r0;
     const Register xlen  = r1;
     const Register z     = r2;
-    const Register zlen  = r3;
     const Register y     = r4; // == x
     const Register ylen  = r5; // == xlen
 
-    const Register tmp1  = r10;
-    const Register tmp2  = r11;
-    const Register tmp3  = r12;
-    const Register tmp4  = r13;
-    const Register tmp5  = r14;
-    const Register tmp6  = r15;
-    const Register tmp7  = r16;
+    const Register tmp0  = r10;
+    const Register tmp1  = r11;
+    const Register tmp2  = r12;
+    const Register tmp3  = r13;
+    const Register tmp4  = r14;
+    const Register tmp5  = r15;
+    const Register tmp6  = r16;
+    const Register tmp7  = r17;
```

Comment:
No need for r17 or sorting tmps.  Make tmp0 r3, or r6, r7, etc.

Reply:
Also, I don't see why the code below saves and restores r4/r5.  Maybe @theRealAph knows?  Aren't all these registers killed across a runtime call?